app/exchanges/volume_tracking/message_builder.rb

```diff
@@ -56,8 +57,13 @@ def populate_by_source_type(data)
         data[:source_type] = 'pool'
         data[:source_barcode] = object.source.tube.barcode
         data[:sample_name] = pacbio_library_sample_names
+      when 'Pacbio::Request'
+        data[:source_type] = 'request'
+        data[:source_barcode] = object.source.container.barcode
```

Comment:
Oh,  Is there a source.container accessor? Good way to get source barcode which avoids checking whether it's a tube or a plate. 👍

Reply:
Courtesy of @BenTopping 😄

Reply:
@BenTopping 🔥 😄

Reply:
Technically the container is either a well or a tube, but well's delegate barcode to their plate. So you can use it to retrieve the barcode.